reconcile/utils/saasherder/saasherder.py

```diff
@@ -1877,82 +1878,101 @@ def sanitize_namespace(
+        # Placing this check here to make mypy happy
         if not (self.state and self._promotion_state):
             raise Exception("state is not initialized")
 
-        for promotion in self.promotions:
-            if promotion is None:
-                continue
-            # validate that the commit sha being promoted
-            # was successfully published to the subscribed channel(s)
-            if promotion.subscribe:
-                for channel in promotion.subscribe:
-                    config_hashes: set[str] = set()
-                    for target_uid in channel.publisher_uids:
-                        deployment = self._promotion_state.get_promotion_data(
-                            sha=promotion.commit_sha,
-                            channel=channel.name,
-                            target_uid=target_uid,
-                            local_lookup=False,
-                        )
-                        if not (deployment and deployment.success):
-                            logging.error(
-                                f"Commit {promotion.commit_sha} was not "
-                                + f"published with success to channel {channel.name}"
-                            )
-                            return False
-                        if deployment.target_config_hash:
-                            config_hashes.add(deployment.target_config_hash)
-
-                    # This code supports current saas targets that does
-                    # not have promotion_data yet
-                    if not config_hashes or not promotion.promotion_data:
-                        logging.info(
-                            "Promotion data is missing; rely on the success "
-                            "state only"
-                        )
-                        return True
-
-                    # Validate the promotion_data section.
-                    # Just validate parent_saas_config hash
-                    # promotion_data type by now.
-                    parent_saas_config = None
-                    for pd in promotion.promotion_data:
-                        if pd.channel == channel.name:
-                            for data in pd.data or []:
-                                if isinstance(data, SaasParentSaasPromotion):
-                                    parent_saas_config = data
-
-                    # This section might not exist due to a manual MR.
-                    # Promotion shall continue if this data is missing.
-                    # The parent at the same ref has succeed if this code
-                    # is reached though.
-                    if not parent_saas_config:
-                        logging.info(
-                            "Parent Saas config missing on target "
-                            "rely on the success state only"
-                        )
-                        return True
+        if not promotion:
+            return True
```

Comment:
note you can mark `promotion` as `Promotion` to only support not None as we already filtered out None values when use.

Reply:
Good catch! Fixed